model/src/cmd/ModelCommandAPI.cc

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "ModelCommandAPI.hh"
+
+#include <string>
+#include <vector>
+#include <map>
+
+#include <ignition/common/Console.hh>
+#include <ignition/common/Filesystem.hh>
+#include <ignition/gazebo/components/ChildLinkName.hh>
+#include <ignition/gazebo/components/Inertial.hh>
+#include <ignition/gazebo/components/Joint.hh>
+#include <ignition/gazebo/components/JointType.hh>
+#include <ignition/gazebo/components/Link.hh>
+#include <ignition/gazebo/components/Model.hh>
+#include <ignition/gazebo/components/Name.hh>
+#include <ignition/gazebo/components/ParentEntity.hh>
+#include <ignition/gazebo/components/ParentLinkName.hh>
+#include <ignition/gazebo/components/Pose.hh>
+#include <ignition/gazebo/EntityComponentManager.hh>
+#include <ignition/msgs.hh>
+#include <ignition/msgs/serialized.pb.h>
+#include <ignition/transport/Node.hh>
+
+namespace {
+
+//////////////////////////////////////////////////
+// \brief Get the name of the world being used by calling
+// `/gazebo/worlds` service.
+// \return The name of the world if service is available,
+// an empty string otherwise.
+std::string getWorldName()
+{
+  // Create a transport node.
+  ignition::transport::Node node;
+
+  bool result{false};
+  const unsigned int timeout{5000};
+  const std::string service{"/gazebo/worlds"};
+
+  // Request and block
+  ignition::msgs::StringMsg_V res;
+
+  if (!node.Request(service, timeout, res, result))
+  {
+    std::cerr << std::endl << "Service call to [" << service << "] timed out"
+              << std::endl;
+    return "";
+  }
+
+  if (!result)
+  {
+    std::cerr << std::endl << "Service call to [" << service << "] failed"
+              << std::endl;
+    return "";
+  }
+
+  return res.data().Get(0);
+}
+
+//////////////////////////////////////////////////
+// \brief Set the state of a ECM instance with a world snapshot.
+// \param _ecm ECM instance to be populated.
+// \return boolean indicating if it was able to populate the ECM.
+bool PopulateECM(ignition::gazebo::EntityComponentManager &_ecm)
```

Comment:
We try to use `camelCase` for free functions, but we don't have that documented anywhere :upside_down_face: 

```suggestion
bool populateECM(ignition::gazebo::EntityComponentManager &_ecm)
```

Reply:
Addressed